aws-codeguruprofiler-profilinggroup/src/main/java/software/amazon/codeguruprofiler/profilinggroup/NotificationChannelHelper.java

```diff
@@ -0,0 +1,58 @@
+package software.amazon.codeguruprofiler.profilinggroup;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.ImmutableSet;
+import software.amazon.awssdk.services.codeguruprofiler.CodeGuruProfilerClient;
+import software.amazon.awssdk.services.codeguruprofiler.model.AddNotificationChannelsRequest;
+import software.amazon.awssdk.services.codeguruprofiler.model.Channel;
+import software.amazon.awssdk.services.codeguruprofiler.model.EventPublisher;
+import software.amazon.awssdk.services.codeguruprofiler.model.RemoveNotificationChannelRequest;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+
+public class NotificationChannelHelper {
+    private NotificationChannelHelper() {
+        // prevent instantiation
+    }
+
+    public static void addChannelNotifications(String pgName, List<software.amazon.codeguruprofiler.profilinggroup.Channel> channels,
+                                               AmazonWebServicesClientProxy proxy, CodeGuruProfilerClient profilerClient) {
+        AddNotificationChannelsRequest.Builder addNotificationChannelsRequest = AddNotificationChannelsRequest.builder()
+                .profilingGroupName(pgName);
+        addNotificationChannelsRequest.channels(channels.stream().map(channel -> Channel.builder().uri(channel.getChannelUri())
+                .eventPublishers(ImmutableSet.of(EventPublisher.ANOMALY_DETECTION))
+                .id(channel.getChannelId()).build()).collect(Collectors.toList()));
+
+        proxy.injectCredentialsAndInvokeV2(addNotificationChannelsRequest.build(), profilerClient::addNotificationChannels);
+    }
```

Comment:
Thanks for fixing the bug I pointed out. Can you also add a test-case that would fail with the older code, but works with the new one? I think this is worth including in the tests.

Reply:
I have added a unit test that verifies multiple channels works correctly. I am hesitant to add a test that asserts the previous test fails just because it would add some test infra setup and for the interest of time.

Reply:
The current test seems enough -- my comment was more along the lines of locally copy pasting the old version of the code, seeing the test fail, and then replacing the fixed code and seeing the test works. Quite a few times I've done so myself and be surprised that the old code didn't fail, because the test was missing something and was incorrect.

Reply:
Ahh gotcha. I actually did that locally and it `did` fail, so good catch :)